nav2_simple_commander/nav2_simple_commander/example_nav_through_poses.py

```diff
@@ -0,0 +1,140 @@
+    goal_pose1.pose.position.x = 1.5
+    goal_pose1.pose.position.y = 0.55
+    goal_pose1.pose.orientation.w = 0.707
+    goal_pose1.pose.orientation.z = 0.707
+    goal_poses.append(goal_pose1)
+
+    # additional goals can be appended
+    goal_pose2 = PoseStamped()
+    goal_pose2.header.frame_id = 'map'
+    goal_pose2.header.stamp = navigator.get_clock().now().to_msg()
+    goal_pose2.pose.position.x = 1.5
+    goal_pose2.pose.position.y = -3.75
+    goal_pose2.pose.orientation.w = 0.707
+    goal_pose2.pose.orientation.z = 0.707
+    goal_poses.append(goal_pose2)
+    goal_pose3 = PoseStamped()
+    goal_pose3.header.frame_id = 'map'
+    goal_pose3.header.stamp = navigator.get_clock().now().to_msg()
+    goal_pose3.pose.position.x = -3.6
+    goal_pose3.pose.position.y = -4.75
+    goal_pose3.pose.orientation.w = 0.707
+    goal_pose3.pose.orientation.z = 0.707
+    goal_poses.append(goal_pose3)
+
+    # sanity check a valid path exists
+    # path = navigator.getPathThroughPoses(initial_pose, goal_poses)
+
+    navigator.goThroughPoses(goal_poses)
+
+    i = 0
+    while not navigator.isNavComplete():
+        ################################################
+        #
+        # Implement some code here for your application!
+        #
+        ################################################
+
+        # Do something with the feedback
+        i = i + 1
+        feedback = navigator.getFeedback()
+        if feedback and i % 5 == 0:
+            print('Estimated time of arrival: ' + '{0:.0f}'.format(
+                  Duration.from_msg(feedback.estimated_time_remaining).nanoseconds / 1e9)
+                  + ' seconds.')
+
+            # Some navigation timeout to demo cancellation
+            if Duration.from_msg(feedback.navigation_time) > Duration(seconds=600.0):
+                navigator.cancelNav()
+
+            # Some navigation request change to demo preemption
+            if Duration.from_msg(feedback.navigation_time) > Duration(seconds=35.0):
+                goal_pose4 = PoseStamped()
+                goal_pose4.header.frame_id = 'map'
+                goal_pose4.header.stamp = navigator.get_clock().now().to_msg()
+                goal_pose4.pose.position.x = -5.0
+                goal_pose4.pose.position.y = -4.75
+                goal_pose4.pose.orientation.w = 0.707
+                goal_pose4.pose.orientation.z = 0.707
+                navigator.goThroughPoses([goal_pose4])
+
+    # Do something depending on the return code
+    result = navigator.getResult()
+    if result == NavigationResult.SUCCEEDED:
+        print('Goal succeeded!')
+    elif result == NavigationResult.CANCELED:
+        print('Goal was canceled!')
+    elif result == NavigationResult.FAILED:
+        print('Goal failed!')
+    else:
+        print('Goal has an invalid return status!')
+
+    navigator.lifecycleShutdown()
+
+    exit(0)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
nit: PEP8 linting will probably fail here because of missing new line at end of file (depending on the flake8 / linting config)